common/djangoapps/course_category/models.py

```diff
@@ -69,7 +69,7 @@ def move_reindex_course_category(sender, instance, **kwargs):
 def delete_reindex_course_category(sender, instance, **kwargs):
     category_courses = instance.courses.all()
     if category_courses:
-        instance.courses_list = (str(x.id) for x in category_courses)
+        instance.courses_list = list((str(x.id) for x in category_courses))
```

Comment:
A list must be created via list comprehension.
`[str(x.id) for x in category_courses]`